ccm_web/server/src/canvas/invalid.token.interceptor.ts

```diff
@@ -0,0 +1,42 @@
+import { Request } from 'express'
+import {
+  CallHandler, ExecutionContext, HttpException, HttpStatus, Injectable, NestInterceptor,
+  UnauthorizedException
+} from '@nestjs/common'
+import { Observable } from 'rxjs'
+import { catchError } from 'rxjs/operators'
+
+import { CanvasTokenNotFoundError, InvalidTokenRefreshError } from './canvas.errors'
+import { CanvasService } from './canvas.service'
+import { RequestWithoutUserError } from '../user/user.errors'
+
+@Injectable()
+export class InvalidTokenInterceptor implements NestInterceptor {
+  constructor (private readonly canvasService: CanvasService) {}
+
+  async intercept (context: ExecutionContext, next: CallHandler): Promise<Observable<unknown>> {
+    const req = context.switchToHttp().getRequest<Request>()
+    const user = req.user
+    if (user === undefined) throw new RequestWithoutUserError()
+
+    return next.handle().pipe(
+      catchError(async (err) => {
+        if (err instanceof CanvasTokenNotFoundError) {
+          throw new UnauthorizedException('The tool does not have a Canvas token for you.')
+        }
```

Comment:
@chrisrrowland, I added this to handle the case you brought up; let me know what you think when you can.